From: Johannes Thumshirn <jthumshirn@suse.de>
Date: Mon, 26 Feb 2018 16:22:36 +0100
Subject: [PATCH] scsi: don't look for NULL devices handlers by name
Patch-Mainline: Never, SLE12 SP3 specific
References: bsc#1082373

Currently scsi_dh_lookup() doesn't check for NULL as a device
name. This combined with nvme over dm-mpath results in the following
messages emitted by device-mapper:

 device-mapper: multipath: Could not failover device 259:67: Handler scsi_dh_(null) error 14.

Let scsi_dh_lookup() fail fast on NULL names.

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/scsi_dh.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/scsi/scsi_dh.c b/drivers/scsi/scsi_dh.c
index 16e780d23375..6059207899f0 100644
--- a/drivers/scsi/scsi_dh.c
+++ b/drivers/scsi/scsi_dh.c
@@ -109,6 +109,9 @@ static struct scsi_device_handler *scsi_dh_lookup(const char *name)
 {
 	struct scsi_device_handler *dh;
 
+	if (!name || strlen(name) == 0)
+		return NULL;
+
 	dh = __scsi_dh_lookup(name);
 	if (!dh) {
 		request_module("scsi_dh_%s", name);
-- 
2.12.3

